src/server/routes/wallet-credentials/get-all.ts

```diff
@@ -0,0 +1,84 @@
+import { Type, type Static } from "@sinclair/typebox";
+import type { FastifyInstance } from "fastify";
+import { StatusCodes } from "http-status-codes";
+import { getAllWalletCredentials } from "../../../shared/db/wallet-credentials/get-all-wallet-credentials";
+import { standardResponseSchema } from "../../schemas/shared-api-schemas";
+
+const QuerySchema = Type.Object({
+  page: Type.Integer({
+    description: "The page of credentials to get.",
+    examples: ["1"],
+    default: "1",
+    minimum: 1,
+  }),
+  limit: Type.Integer({
+    description: "The number of credentials to get per page.",
+    examples: ["10"],
+    default: "10",
+    minimum: 1,
+  }),
+});
+
+const responseSchema = Type.Object({
+  result: Type.Array(
+    Type.Object({
+      id: Type.String(),
+      type: Type.String(),
+      label: Type.Union([Type.String(), Type.Null()]),
+      isDefault: Type.Boolean(),
+      createdAt: Type.String(),
+      updatedAt: Type.String(),
+      deletedAt: Type.Union([Type.String(), Type.Null()]),
+    }),
+  ),
+});
+
+responseSchema.example = {
+  result: [
+    {
+      id: "123e4567-e89b-12d3-a456-426614174000",
+      type: "circle",
+      label: "My Circle Credential",
+      isDefault: false,
+      createdAt: "2024-01-01T00:00:00.000Z",
+      updatedAt: "2024-01-01T00:00:00.000Z",
+      deletedAt: null,
+    },
+  ],
+};
+
+export async function getAllWalletCredentialsRoute(fastify: FastifyInstance) {
+  fastify.route<{
+    Querystring: Static<typeof QuerySchema>;
+    Reply: Static<typeof responseSchema>;
+  }>({
+    method: "GET",
+    url: "/wallet-credentials",
+    schema: {
+      summary: "Get all wallet credentials",
+      description: "Get all wallet credentials with pagination.",
+      tags: ["Wallet Credentials"],
+      operationId: "getAllWalletCredentials",
+      querystring: QuerySchema,
+      response: {
+        ...standardResponseSchema,
+        [StatusCodes.OK]: responseSchema,
+      },
+    },
+    handler: async (req, res) => {
+      const credentials = await getAllWalletCredentials({
+        page: req.query.page,
+        limit: req.query.limit,
+      });
+
+      res.status(StatusCodes.OK).send({
+        result: credentials.map((cred) => ({
```

Comment:
credentials includes the entitySecret here which I think you don't want to return.

Suggestion: create a `toWalletCredentialSchema()` helper function ([example](https://github.com/thirdweb-dev/engine/blob/main/src/server/schemas/webhook.ts#L14-L24)) that maps the Prisma type to the schema type (doing any transformations and omitting sensitive details), then reuse this function in any get/list handlers.

```typescript
result: credentials.map(toWalletCredentialSchema)
```

Reply:
`getAllWalletCredentials` skips the secret at the DB query:
```

  const credentials = await prisma.walletCredentials.findMany({
    where: {
      deletedAt: null,
    },
    skip: (page - 1) * limit,
    take: limit,
    select: {
      id: true,
      type: true,
      label: true,
      isDefault: true,
      createdAt: true,
      updatedAt: true,
      deletedAt: true,
    },
    orderBy: {
      createdAt: "desc",
    },
  });
  ```

Reply:
Ah, getAllWalletCredentials omits it but getWalletCredentials provides it.